app/src/main/java/org/wikipedia/edit/summaries/EditSummaryFragment.kt

```diff
@@ -1,82 +1,179 @@
+    private lateinit var localizeSummaryTags: SparseArray<String>
     lateinit var title: PageTitle
 
-    /**
-     * Gets the custom ("other") summary, if any, that the user has entered.
-     * @return Custom summary of the edit.
-     */
-    val summary get() = binding.editSummaryEdit.text.toString()
+    val summary get() = binding.editSummaryText.text.toString()
+    val isMinorEdit get() = binding.minorEditCheckBox.isChecked
+    val watchThisPage get() = binding.watchPageCheckBox.isChecked
     val isActive get() = binding.root.visibility == View.VISIBLE
 
+    private val summaryTagStrings = intArrayOf(R.string.edit_summary_tag_typo, R.string.edit_summary_tag_grammar, R.string.edit_summary_tag_links)
+
+    private val voiceSearchLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+        val voiceSearchResult = it.data?.getStringArrayListExtra(RecognizerIntent.EXTRA_RESULTS)
+        if (it.resultCode == Activity.RESULT_OK && voiceSearchResult != null) {
+            val text = voiceSearchResult.first()
+            binding.editSummaryText.setText(text)
+        }
+    }
+
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
         _binding = FragmentPreviewSummaryBinding.inflate(layoutInflater, container, false)
 
         // Explicitly enable standard dictionary autocompletion in the edit summary box
         // We should be able to do this in the XML, but doing it there doesn't work. Thanks Android!
-        binding.editSummaryEdit.inputType = binding.editSummaryEdit.inputType and EditorInfo.TYPE_TEXT_FLAG_AUTO_COMPLETE.inv()
+        binding.editSummaryText.inputType = binding.editSummaryText.inputType and EditorInfo.TYPE_TEXT_FLAG_AUTO_COMPLETE.inv()
 
         // ...so that clicking the "Done" button on the keyboard will have the effect of
         // clicking the "Next" button in the actionbar:
-        binding.editSummaryEdit.setOnEditorActionListener { _, actionId, keyEvent ->
+        binding.editSummaryText.setOnEditorActionListener { _, actionId, keyEvent ->
             if ((keyEvent != null && keyEvent.keyCode == KeyEvent.KEYCODE_ENTER) || actionId == EditorInfo.IME_ACTION_DONE) {
                 (requireActivity() as EditSectionActivity).clickNextButton()
             }
             false
         }
 
-        if (savedInstanceState != null) {
-            binding.editSummaryEdit.setText(savedInstanceState.getString(KEY_SUMMARY_TEXT))
+        binding.editSummaryText.addTextChangedListener {
+            if (!requireActivity().isDestroyed) {
+                requireActivity().invalidateOptionsMenu()
+            }
+        }
+
+        binding.editSummaryTextLayout.setEndIconOnClickListener {
+            launchVoiceInput()
+        }
+
+        binding.learnMoreButton.setOnClickListener {
+            CustomTabsUtil.openInCustomTab(requireContext(), getString(R.string.preview_edit_learn_more_url))
```

Comment:
It doesn't look like CustomTabs will work for these cases. Because these are Wikipedia URLs, it still brings up the default app-chooser, and lets the user select the Wikipedia app. If they select Wikipedia, it will launch PageActivity, and will erase the current edit that the user was making.

These URLs should probably be handled by calling `visitInExternalBrowser()`.

Reply:
Makes sense. I'll let Robin know about this. Thanks!